packages/hardhat-core/src/internal/core/project-structure.ts

```diff
@@ -9,11 +10,17 @@ import { ERRORS } from "./errors-list";
 const JS_CONFIG_FILENAME = "hardhat.config.js";
 const TS_CONFIG_FILENAME = "hardhat.config.ts";
 
+const log = debug("hardhat:core:prj-structure");
+
 export function isCwdInsideProject() {
-  return (
-    findUp.sync(TS_CONFIG_FILENAME) !== null ||
-    findUp.sync(JS_CONFIG_FILENAME) !== null
-  );
+  const jsConfigPath = findUp.sync(JS_CONFIG_FILENAME);
+  const tsConfigPath = findUp.sync(TS_CONFIG_FILENAME);
```

Comment:
Maybe a good idea to relocate *ConfigPath vars out of functions as getUserConfigPath() contains the same vars.